src/main/java/ch/cern/eam/wshub/core/services/entities/Pair.java

```diff
@@ -47,4 +48,25 @@ public String toString() {
 		return map;
 	}
 
+	@Override
+	public boolean equals(Object other){
+		if(other instanceof Pair){
+			Pair otherPair = (Pair) other;
+			boolean codeEq = otherPair.getCode() == code ||
+					otherPair.getCode() != null && code != null && otherPair.getCode().equals(code);
+			boolean descEq = otherPair.getDesc() == desc ||
+					otherPair.getDesc()  != null && desc != null && otherPair.getDesc().equals(desc);
+			return codeEq && descEq;
+		}
+		return false;
+	}
+
+	@Override
+	public int hashCode(){
```

Comment:
This implementation works, but I believe it would be a cleaner solution if we do the following:
```java
return Objects.hash(code, desc);
```
see https://docs.oracle.com/javase/7/docs/api/java/util/Objects.html#hash(java.lang.Object...)